src/main/java/io/confluent/kafka/connect/datagen/DatagenTask.java

```diff
@@ -66,7 +67,7 @@
   private AvroData avroData;
   private int taskId;
   private Map<String, Object> sourcePartition;
-  private int taskGeneration;
+  private long taskGeneration;
```

Comment:
This variable is only ever initialized if a prior source offset for the task has been stored. Can we add an initializer either here or in the task constructor to set it to zero if no prior offset has been found? Technically this works with Java's default values for non-final instance variables but at least for me personally it'd help with readability to make that explicit.